src/app/features/login/login-dialog.component.ts

```diff
@@ -36,10 +42,40 @@ export class LoginDialog implements ModalComponent<LoginDialogParameters> {
 		if (document.body.classList.contains('modal-open')) {
 			document.body.classList.remove('modal-open');
 		}
-		this.dialog.close({
-			username: this.parameters.username,
-			password: this.parameters.password,
-			server:   this.parameters.server
+
+		this.checkConnection().subscribe(() => {
+			this.dialog.close({
+				isLogged: true,
+				username: this.parameters.username,
```

Comment:
These assignments are already done at the checkConnection() method. I think these 3 lines can be removed.

Reply:
I think it should to return the object with all the fields because app.component has a subscription to get this object and set its own username and password variables to use in this component.